libs/zgpu/src/binding_tests.zig

```diff
@@ -0,0 +1,84 @@
+const std = @import("std");
+const testing = std.testing;
+const c = @cImport({
+    @cInclude("webgpu/webgpu.h");
+});
+const gpu = @import("wgpu.zig");
+
+// test struct sizes
+fn assertStructBindings(zig_type: anytype, c_type: anytype) void {
+    if (@sizeOf(zig_type) != @sizeOf(c_type)) {
+        @compileLog("emscripten zgpu type has different size from webgpu.h type:", zig_type, c_type, @sizeOf(zig_type), @sizeOf(c_type));
+        unreachable;
+    }
+    // more checks? iterate struct fields?
+    // emscripten in .Debug mode asserts at runtime if required field is missing etc. but something always slips through
```

Comment:
core issue is why I added these tests is that `webgpu.zig` basically rewrites c header `webgpu.h`. By default it matches dawn `webgpu.h` but emscripten ships with its own that can have version mismatch. If zig structs doesn't match with ones in actual header problems start. For emscripten that means it will try read some struct field but might get value from different field etc. Size checks help to quickly at comptime show these. But if u32 field was removed to struct and other added at end, it will slip through leading to unknown behavior and most likely crash at runtime or worse. But its not straight forward to test at comptime whole struct due to camel-case/snake-case naming changes.

Reply:
@michal-z do you have any thoughts regarding how struct layouts can be validated?